src/gui/render.cpp

```diff
@@ -624,71 +623,106 @@ static void ChangeScaler(bool pressed) {
 } */
 
 #if C_OPENGL
-static bool RENDER_GetShader(std::string &shader_path, char *old_src)
+
+// Reads the given shader path into the string
+static bool read_shader(const std_fs::path &shader_path, std::string &shader_str)
 {
-	const std::unordered_map<std::string, const char *> builtin_shaders = {
-	        {"advinterp2x", advinterp2x_glsl},
-	        {"advinterp3x", advinterp3x_glsl},
-	        {"advmame2x", advmame2x_glsl},
-	        {"advmame3x", advmame3x_glsl},
-	        {"crt-easymode-flat", crt_easymode_tweaked_glsl},
-	        {"crt-fakelottes-flat", crt_fakelottes_tweaked_glsl},
-	        {"default", sharp_glsl},
-	        {"rgb2x", rgb2x_glsl},
-	        {"rgb3x", rgb3x_glsl},
-	        {"scan2x", scan2x_glsl},
-	        {"scan3x", scan3x_glsl},
-	        {"sharp", sharp_glsl},
-	        {"tv2x", tv2x_glsl},
-	        {"tv3x", tv3x_glsl},
-	};
-
-	char* src;
-	std::stringstream buf;
-	std::ifstream fshader(shader_path.c_str(), std::ios_base::binary);
+	std::ifstream fshader(shader_path, std::ios_base::binary);
 	if (!fshader.is_open())
-		fshader.open(shader_path + ".glsl", std::ios_base::binary);
-	if (fshader.is_open()) {
-		buf << fshader.rdbuf();
-		fshader.close();
-	} else if (builtin_shaders.count(shader_path) > 0) {
-		buf << builtin_shaders.at(shader_path);
+		return false;
+
+	std::stringstream buf;
+	buf << fshader.rdbuf();
+	fshader.close();
+	if (buf.str().empty())
+		return false;
+
+	shader_str = buf.str();
+	shader_str += '\n';
+	return true;
+}
+
+std::deque<std::string> RENDER_InventoryShaders()
+{
```

Comment:
Would be useful to show the user the empty `glshader` directories too, so they're aware of what options exist (for provide their own shaders).